### DISCUSSION TITLE
Fixing an improper method call

### DISCUSSION THREAD

**Discussion by nydelzaf** [SciTools/iris#5521] [General]
In file: [coords.py](https://github.com/SciTools/iris/blob/main/lib/iris/coords.py#L3122), class: CellMethod, there is a special method __add__ that raises a NotImplementedError. If a special method supporting a [binary operation is not implemented it should return NotImplemented](https://docs.python.org/3/library/constants.html#NotImplemented). On the other hand, NotImplementedError should be raised from abstract methods inside user defined base classes to indicate that [derived classes should override those methods. ](https://docs.python.org/3/library/exceptions.html#NotImplementedError)iCR suggested that the special method __add__ should return NotImplemented instead of raising an exception.

An example of how NotImplemented helps the [interpreter support a binary operation is here.](https://docs.python.org/3/library/numbers.html#implementing-the-arithmetic-operations)

So, the changes to the code would be something like-
```diff
def __add__(self, other):
  # Disable the default tuple behaviour of tuple concatenation
- raiseNotImplementedError()
+ returnNotImplemented
```

I see the comment written above the exception, but still wondering if it was meant for using `NotImplementedError` instead of `NotImplemented` or not. If it's the later, then I suggest that we follow Python's guideline and use `return NotImplemented`

If it's alright, I can create a PR that you can merge.

Looking forward to hearing from you.

<br><br>

**Sponsorship and Support:**

This work is done by the security researchers from OpenRefactory and is supported by the [Open Source Security Foundation (OpenSSF)](https://openssf.org/): [Project Alpha-Omega](https://alpha-omega.dev/). Alpha-Omega is a project partnering with open source software project maintainers to systematically find new, as-yet-undiscovered vulnerabilities in open source code - and get them fixed – to improve global software supply chain security.

The bug is found by running the Intelligent Code Repair (iCR) tool by OpenRefactory and then manually triaging the results.

**Comment by trexfeathers** [comment]
Hi @fazledyn-or

First of all: thanks so much for checking our code base. What a great idea for making the world a better place™ ❤

We discussed as a team this week and we _think_ `NotImplemented` is the right way to go. It's hard to be certain since [this line has been in the code forever](https://github.com/SciTools/iris/blame/main/lib/iris/coords.py#L3120-L3122), but I ran the tests with `NotImplemented` and everything passed.

If you'd like the credit of raising a pull request, please go ahead. If you're busy, we'll get to it in the near future.

**Reply by nydelzaf** [reply]
Great! Let me submit a PR then.

**Reply by nydelzaf** [reply]
Please have a look at it. PR https://github.com/SciTools/iris/pull/5544
